tests/dateSelector.spec.ts

```diff
@@ -0,0 +1,140 @@
+    //4. In the Name field, type any new pet name, for example "Tom"
+    const icon = page.locator('span.glyphicon.form-control-feedback').first()
+    //assert the icon 'x'
+    await expect(icon).toHaveClass(/glyphicon-remove/)
+    await page.locator('#name').fill('Tom')
+    //5. Add the assertion of icon in the input field the "checkmark" icon
+    await expect(icon).toHaveClass(/glyphicon-ok/)
+    //6. Click on the calendar icon for the "Birth Date" field
+    await page.getByLabel('Open calendar').click()
+    //7. Using calendar selector, select the date "May 2nd, 2014"
+    const birthYear = '2014';
+    const birthMonth = '05';
+    const birthDay = '02';
+    const dateToAssert = `${birthYear}/${birthMonth}/${birthDay}`
+
+    await page.getByLabel('Choose month and year').click()
+    await page.getByLabel('Previous 24 years').click()
+    await page.getByRole('button', { name: birthYear }).click()
+    await page.getByLabel(`${birthMonth} ${birthYear}`).click()
+    await page.getByLabel(dateToAssert).click()
+    await expect(page.locator('[name="birthDate"]')).toHaveValue(dateToAssert)
+    //9. Select the type of pet "dog" and click "Save Pet" button
+    await page.getByLabel('Type').selectOption('dog')
+    await page.getByRole('button', { name: 'Save Pet' }).click()
+    //10. On the Owner Information page, add assertions for the newly created pet. Name is Tom, Birth Date is in the format "2014-05-02", Type is dog
+    const petField = page.locator('td', { has: page.getByText('Tom') })
+    await expect(petField).toContainText(`${birthYear}-${birthMonth}-${birthDay}`)
+    await expect(petField).toContainText('dog')
+    //11. Click "Delete Pet" button the for the new pet "Tom"
+    await page.getByRole('button', { name: 'Delete Pet' }).last().click()
+    //12. Add assertion that Tom does not exist in the list of pets anymore
+    await expect(petField).toHaveCount(0)
+})
+test('select the desired date in the calendar part II', async ({ page }) => {
+    // 2. In the list of the Owners, locate the owner by the name "Jean Coleman" and select this owner
+    await page.getByRole('link', { name: 'Jean Coleman' }).click()
+    //3. In the list of pets, locate the pet with a name "Samantha" and click "Add Visit" button
+    const addVisitButtonOwnerPage = page.locator('app-pet-list', { hasText: 'Samantha' })
+    await addVisitButtonOwnerPage.getByText('Add Visit').click()
+    //4. Add the assertion that "New Visit" is displayed as header of the page
+    await expect(page.getByRole("heading")).toHaveText("New Visit")
+    //5. Add the assertion that pet name is "Samantha" and owner name is "Jean Coleman"
+    await expect(page.locator('tr', { hasText: 'Samantha' })).toContainText('Jean Coleman')
+    //6. Click on the calendar icon and select the current date in date picker
+    const calendarIcon = page.getByLabel('Open calendar')
+    await calendarIcon.click()
```

Comment:
`calendarIcon` constant was not used anywhere else in the code, so can be removed

Reply:
make sense